src/frontends/pytorch/src/transforms/fftn_complex_replacer.cpp

```diff
@@ -0,0 +1,151 @@
+        } else {
+            // Dim and s are set to default, use all of dimensions.
+            dim = std::make_shared<v4::Range>(const_0, input_rank_scalar, const_1, element::i32);
+        }
+
+        // Handle s parameter containing vector of intigers indicating signal sizes for dimensions.
+        std::shared_ptr<ov::Node> s;
+        if (!s_use_default) {
+            // Values for s were provided. Replace -1 values with default full size in given dimension.
+            auto full_s_cond = std::make_shared<v1::Equal>(raw_s_input_maybe, const_neg_1);
+            auto full_s_values = std::make_shared<v8::Gather>(input_shape, dim, const_0);
+            s = std::make_shared<v1::Select>(full_s_cond, full_s_values, raw_s_input_maybe);
+        } else {
+            // Value for s was set to default, use full size for all dimensions.
+            s = std::make_shared<v8::Gather>(input_shape, dim, const_0);
+        }
+
+        // Handle norm parameter indicating normalization mode to use. Defaults to "backward".
+        std::string norm;
+        if (const auto& fw_node_mode =
+                std::dynamic_pointer_cast<ov::op::util::FrameworkNode>(fftn_op->input_value(3).get_node_shared_ptr())) {
+            const auto& attrs = fw_node_mode->get_attrs();
+            if (attrs.find("string_value") != attrs.end()) {
+                norm = attrs.at("string_value");
+            } else {
+                norm = "backward";
+            }
+        } else {
+            add_exception_to_fw_node(fftn_op, "aten::fft_fftn: could not retrive value for norm attribute.");
+            return false;
+        }
+
+        auto rdft = std::make_shared<v9::RDFT>(input, dim, s);
+
+        // Apply normalizations
+        auto n_int = std::make_shared<v1::ReduceProd>(s, const_0);
+        auto n = std::make_shared<v1::ConvertLike>(n_int, rdft);
+        std::shared_ptr<ov::Node> normalized_fftn;
+        if (norm == "forward") {
+            // Normalize by 1/n
+            normalized_fftn = std::make_shared<v1::Divide>(rdft, n);
+        } else if (norm == "backward") {
+            // No normalization
+            normalized_fftn = rdft;
+        } else if (norm == "ortho") {
+            // Normalize by 1/sqrt(n)
+            auto sqrt_n = std::make_shared<v0::Sqrt>(n);
+            normalized_fftn = std::make_shared<v1::Divide>(rdft, sqrt_n);
+        } else {
+            add_exception_to_fw_node(
+                fftn_op,
+                "aten::fft_fftn: unrecognized normalization mode. Only forward, backward and ortho are supported.");
+            return false;
+        }
+
+        // Replace outputs that are either torch operators aten::real or aten::imag. Apply squeeze to remove last
+        // dimension used to concatenate.
+        auto fftn_outs = fftn_op->get_users();
+        bool rval = false;
+        for (auto& out : fftn_outs) {
```

Comment:
I think, the check output operation, should be before before transformation (where you match operation type) and returns false for case if different operation used.

It is not recommended to raise exception in transformation, as it may block other transformations applying in case if some specific transformation failed, but there is another that support such case)

Reply:
Hey @eaidova I used #17999 as reference, should we make proposed changes there as well?

Reply:
@mmikolajcz do you have some specific reasons to check that operation result is aten::real after making operation processing? 

@siddhant-0707 I see also a lot of common between transformations, thanks for high-lighting that, can we reuse some code e.g. for the normalization part?

Reply:
Hi @eaidova, @siddhant-0707, if I remember correctly, this workaround was created to 
- support cases where `aten::fft_rfftn` was connected either to only `aten::real`, `aten::imag`, or both of them,
- to replace those nodes with correct split of RDFT output tensor (since real and imag parts are returned concatenated by RDFT, and then need to be splitted to separate tensors).

Probably it could be achieved in other way, by using more complex pattern matching at the beginning and different way to replace outs with correct splits, but I think this was quickest and simplest solution then to enable it.